src/lib/components/DeckGLMap/layers/piechart/pieChartLayer.ts

```diff
@@ -0,0 +1,116 @@
+import { CompositeLayer, COORDINATE_SYSTEM } from "@deck.gl/core";
+import { SolidPolygonLayer, SolidPolygonLayerProps } from "@deck.gl/layers";
+
+export type PieChartLayerProps<D> = SolidPolygonLayerProps<D>;
+
+// INPUT format (PieData):
+// {
+//     "pies": [ {"x": 433600, "y": 6477600, "R": 20, "fractions": [{"value":21, "idx":0}, {"value":13, "idx":1}, {"value":8, "idx":2}]},
+//               {"x": 434556, "y": 6478951, "R": 40, "fractions": [{"value":12, "idx":0}, {"value":18, "idx":1}, {"value":7, "idx":2}]},
+//               {"x": 437086, "y": 6477198, "R": 30, "fractions": [{"value":16, "idx":0}, {"value":20, "idx":1}, {"value":6, "idx":2}]}
+//             ],
+
+//     "properties": [{"color": [255, 0, 0], "label":"oil"}, {"color": [0, 0, 255], "label":"water"}, {"color": [0, 255, 0], "label":"gas"}, {"color": [0, 255, 255], "label":"sand"}]
+// }
+interface PieData {
+    pies: [
+        {
+            x: number;
+            y: number;
+            R: number;
+            fractions: [{ value: number; idx: number }];
+        }
+    ];
+
+    properties: [{ color: [number, number, number]; label: string }];
+}
+
+const defaultProps = {};
+export default class PieChartLayer extends CompositeLayer<
+    unknown,
+    PieChartLayerProps<unknown>
+> {
+    renderLayers(): SolidPolygonLayer<unknown>[] {
+        const properties: PieChartLayerProps<unknown> = {
+            id: "pie-layer",
+            data: makePies(this.props.data as PieData),
+            pickable: true,
+            getFillColor: (d: {
+                properties: { color: [number, number, number]; label: string };
+            }) => d?.properties?.color ?? [0, 0, 0],
+        };
+
+        const layer = new SolidPolygonLayer(this.getSubLayerProps(properties));
+        return [layer];
+    }
+}
+
+PieChartLayer.layerName = "PieChartLayer";
+PieChartLayer.defaultProps = defaultProps;
+
+//================= Local help functions. ==================
+
+function makePies(data: PieData) {
+    if (data.pies === undefined) {
+        return;
+    }
+
+    const polygons: unknown[] = [];
+    for (const pie of data.pies) {
+        const pie_polygons = makePie(pie, data.properties);
+        for (const polygon of pie_polygons) {
+            polygons.push(polygon);
+        }
+    }
+    return polygons;
+}
+
+// return array of one pie's polygon's
+function makePie(pie, properties) {
```

Comment:
To add types to this function arguments, you would need to extract `Pie` and `PieSliceProps` from `PieData` into separate types, and reuse them both in `PieData` and here. A bit more involved, so I guess it can be changed later.
We can easily add the return type:
```suggestion
function makePie(pie, properties): Position[] {
```